hack/generator/pkg/codegen/pipeline_mark_storage_version.go

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package codegen
+
+import (
+	"context"
+	"sort"
+
+	"github.com/Azure/k8s-infra/hack/generator/pkg/astmodel"
+	"github.com/pkg/errors"
+)
+
+// exportPackages creates a PipelineStage to export our generated code as a set of packages
+func markStorageVersion() PipelineStage {
+	return MakePipelineStage(
+		"markStorageVersion",
+		"Marking the latest version of type as the storage version",
+		func(ctx context.Context, types astmodel.Types) (astmodel.Types, error) {
+			updatedDefs, err := MarkLatestResourceVersionsForStorage(types)
+			if err != nil {
+				return nil, errors.Wrapf(err, "unable to mark latest resource versions for as storage versions")
+			}
+
+			return updatedDefs, nil
+		})
+}
+
+// MarkLatestResourceVersionsForStorage marks the latest version of each resource as the storage version
+func MarkLatestResourceVersionsForStorage(types astmodel.Types) (astmodel.Types, error) {
+
+	result := make(astmodel.Types)
+	resourceLookup, err := groupResourcesByVersion(types)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, def := range types {
+		// see if it is a resource
+		if resourceType, ok := def.Type().(*astmodel.ResourceType); ok {
+
+			unversionedName, err := getUnversionedName(def.Name())
+			if err != nil {
+				// should never happen as all resources have versioned names
+				return nil, err
+			}
+
+			allVersionsOfResource := resourceLookup[unversionedName]
+			latestVersionOfResource := allVersionsOfResource[len(allVersionsOfResource)-1]
+
+			thisPackagePath := def.Name().PackageReference.PackagePath()
+			latestPackagePath := latestVersionOfResource.Name().PackageReference.PackagePath()
+
+			// mark as storage version if it's the latest version
+			isLatestVersion := thisPackagePath == latestPackagePath
+			if isLatestVersion {
+				def = astmodel.MakeTypeDefinition(def.Name(), resourceType.MarkAsStorageVersion()).
+					WithDescription(def.Description())
+			}
+
+			result.Add(def)
+		} else {
+			// otherwise simply add it
+			result.Add(def)
+		}
```

Comment:
Since the **true** branch is overwriting `def`, and `result.Add(def)` is the last statement in both branches, you could move it out of both branches and get rid of the **false** branch completely.

Reply:
Good eye -- done